From d006c4d04aecbe80469a26a6114b776e9de4e3c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADt=20Ondruch?= <vondruch@redhat.com>
Date: Mon, 19 Nov 2012 14:39:19 +0100
Subject: [PATCH] Fix s390x build.

---
 ext/tk/extconf.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/tk/extconf.rb b/ext/tk/extconf.rb
index 8f3bff8..69e90a1 100644
--- a/ext/tk/extconf.rb
+++ b/ext/tk/extconf.rb
@@ -114,7 +114,7 @@ def is_macosx?
 end
 
 def maybe_64bit?
-  /64|universal/ =~ RUBY_PLATFORM
+  /64|universal|s390x/ =~ RUBY_PLATFORM
 end
 
 def check_tcltk_version(version)
-- 
1.8.1

