From 2e15c6bdac1c145cce0c21677477ced8df26718b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADt=20Ondruch?= <vondruch@redhat.com>
Date: Fri, 8 Feb 2013 18:20:50 +0100
Subject: [PATCH] Add configuration arch specific dir flag.

---
 configure.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.in b/configure.in
index d6af000..1c094a3 100644
--- a/configure.in
+++ b/configure.in
@@ -3217,6 +3217,10 @@ AC_SUBST(rubylibprefix)
 rubylibdir='${rubylibprefix}/${ruby_version}'
 rubyarchdir=${multiarch+'${rubyarchprefix}/${ruby_version}'}${multiarch-'${rubylibdir}/${arch}'}
 
+AC_ARG_WITH(rubyarchdir,
+     AS_HELP_STRING([--with-rubyarchdir=DIR], [architecture specific ruby libraries [[LIBDIR/RUBY_BASE_NAME/ARCH]]]),
+            [rubyarchdir=$withval])
+
 rubyarchprefix=${multiarch+'${archlibdir}/${RUBY_BASE_NAME}'}${multiarch-'${rubylibprefix}/${arch}'}
 AC_ARG_WITH(rubyarchprefix,
 	    AS_HELP_STRING([--with-rubyarchprefix=DIR],
-- 
1.8.1.2

