* Unmerged path ChangeLog
diff --git a/configure.in b/configure.in
index be7d4fe..2d511e5 100644
--- a/configure.in
+++ b/configure.in
@@ -2597,14 +2597,9 @@ AS_CASE(["$target_os"],
 
 AC_SUBST(setup)
 
-test x"$prefix" = xNONE && prefix=$ac_default_prefix
-test x"${exec_prefix}" = xNONE && exec_prefix="$prefix"
-pat=`echo "${exec_prefix}" | tr -c '\012' .`'\(.*\)'
-for var in bindir libdir; do
-    eval val='"$'$var'"'
-    AS_CASE(["$val"], ["${exec_prefix}"*], [val='${exec_prefix}'"`expr \"$val\" : \"$pat\"`"])
-    eval $var='"$val"'
-done
+if test "$prefix" = NONE; then
+  prefix=$ac_default_prefix
+fi
 
 BTESTRUBY='$(MINIRUBY)'
 if test x"$cross_compiling" = xyes; then
@@ -3244,9 +3239,6 @@ AS_CASE(["$target_os"],
 
 shvar_to_cpp() {
     local var="$1" val="$2"
-    local exec_prefix_pat="`echo \"${exec_prefix}\" | sed 's/\\./\\\\./g'`"
-    local arch_pat="`echo \"${arch}\" | sed 's/\\./\\\\./g'`"
-    local sitearch_pat="`echo \"${sitearch}\" | sed 's/\\./\\\\./g'`"
     val="`echo '"'"${val}"'"' |
 	sed \
 	    -e 's/\${\([[A-Z][A-Z_]]*\)}/"\1"/g' \
@@ -3269,9 +3261,6 @@ shvar_to_cpp() {
 	    -e 's/\${rubyarchprefix}/"RUBY_ARCH_PREFIX_FOR(arch)"/g' \
 	    -e 's/\${rubysitearchprefix}/"RUBY_SITEARCH_PREFIX_FOR(arch)"/g' \
 	    -e 's/\${exec_prefix}/"RUBY_EXEC_PREFIX"/g' \
-	    -e "s|${exec_prefix_pat}/|"'"RUBY_EXEC_PREFIX"/|g' \
-	    -e "s|${arch_pat}|"'"arch"|g' \
-	    -e "s|${sitearch_pat}|"'"sitearch"|g' \
 	    -e 's|^\"NONE/|RUBY_EXEC_PREFIX\"/|' \
 	    -e 's|^\"NONE\"|\"'"${prefix}"'\"|' \
 	    -e 's/^\"\"\(.\)/\1/;s/\(.\)\"\"$/\1/'
